### PR TITLE
Add reply option to Notifications

### PR DESCRIPTION
**Pull request by priyank-p**
Will work well on all platforms.

To Test it out Toggle DevTools of the active tab and drop this in
```javascript
Notification.onreply = (reply) => {
  console.log('Reply form notification: ', reply);
}

new Notification('Testing Notification', {
  body: 'please reply'
})
```
 Note that you can also use `addEventListener`
```javascript
Notification.addEventListener('reply', (reply) => { 
  console.log('Reply form notification: ', reply);
});
```

### CODE REVIEW

**Comment by priyank-p** [comment]
Seems like it does not work on windows. Found lots of Issues on node-notifier but they did not help seems like need to use alternative` electron-windows-notifications` modules.

Native Notifcation do now work either as i tried them.

**Comment by priyank-p** [comment]
Ready for MacOs and Linux, need to figure something else out for windows since there are lots of issues open in Electron, node-notifier for notification not working for windows :/

File: app/renderer/js/notification.js [hunk 1]
```diff
@@ -2,33 +2,139 @@
...
+			const notification = new DefaultNotification(title, opts);
+			notification.addEventListener('click', () => {
+				ipcRenderer.send('focus-app');
+			});
+			return;
+		}
+
+		// focus app incase notification is closed using
+		// close or reply button
+		ipcRenderer.send('focus-app');
```
Comment (akashnimare) [inline app/renderer/js/notification.js]: I don't think we need this here.

File: package.json [hunk 2]
```diff
@@ -53,7 +54,8 @@
         "zip",
         "AppImage"
       ],
-      "maintainer": "Akash Nimare <svnitakash@gmail.com>"
+      "maintainer": "Akash Nimare <svnitakash@gmail.com>",
+      "asarUnpack": "./app/node_modules/node-notifier/vendor/**"
```
Comment (akashnimare) [inline package.json]: Do we really need to unpack this on Linux and Windows?

Comment (priyank-p) [reply package.json]: Yep, The node-notifier says we need to unpack them.

File: package.json [hunk 3]
```diff
@@ -42,7 +42,8 @@
     ],
     "copyright": "©2017 Kandra Labs, Inc.",
     "mac": {
-      "category": "public.app-category.productivity"
+      "category": "public.app-category.productivity",
+      "asarUnpack": "./app/node_modules/node-notifier/vendor/**"
```
Comment (akashnimare) [inline package.json]: @cPhost do we need to unpack this for `node-mac-notifier` also?

Comment (priyank-p) [reply package.json]: No we do need to.

File: app/renderer/js/notification/index.js [hunk 4]
```diff
@@ -4,15 +4,18 @@ const {
   remote: { app }
 } = require('electron');
 
+const { appId } = require('./helpers');
```
Comment (priyank-p) [inline app/renderer/js/notification/index.js]: Thanks :)

**Pull request merged**